client/modules/IDE/components/FileNode.jsx

```diff
@@ -77,6 +78,12 @@ export class FileNode extends React.Component {
     ) {
       this.props.updateFileName(this.props.id, this.originalFileName);
     }
+    if (newFileName === '') {
```

Comment:
Since the each clause of these four `if` statements have the same effect, you could combine them into a single statement. 

Maybe assign the conditionals to variables, like:

```js
const hasNoExtension = oldFileExtension && !newFileExtension;
const hasSameExtension = oldFileExtension && newFileExtension && oldFileExtension[0].toLowerCase() !== newFileExtension[0].toLowerCase();
const hasEmptyFilename = newFileName === '';
const hasOnlyExtension = newFileName === newFileExtension[0];

if (hasNoExtension || hasSameExtension || hasEmptyFilename || hasOnlyExtension) {
  this.props.updateFileName(this.props.id, this.originalFileName);
}
```

Reply:
i like this suggestion! i think it is easier to read and understand.